File: Matlab/SCRoF_mars25/lignesAvec.m
function lign=lignesAvec(mat,val)
%  lign=lignesAvec(mat,val);
% retourne les rangs des lignes de mat qui contiennent un élément de val
nl=size(mat,1);
val(val==0)=[];
lign=1:nl;
for k=nl:-1:1
    A=val(:)==mat(k,:);
    if ~any(A(:))
        lign(k)=[];
    end
end

File: R/lignesAvec.R
lignesAvec <- function(mat, val) {
  # lign=lignesAvec(mat,val);
  # Retourne les rangs des lignes de mat qui contiennent un élément de val
  
  nl <- nrow(mat)
  val <- val[val != 0]  # Supprime les valeurs égales à 0
  lign <- seq_len(nl)
  
  for (k in nl:1) {
    A <- val %in% mat[k, ]  # Vérifie si un élément de val est présent dans la ligne
    if (!any(A)) {
      lign <- lign[lign != k]  # Supprime l'index si aucun élément n'est trouvé
    }
  }
  
  return(lign)
}